js/main.js

```diff
@@ -40,44 +47,74 @@ function renderEntries(entry) {
 }
 
 var $entryForm = document.querySelector('.entry-form');
-var entryId = 0;
 
 $entryForm.addEventListener('submit', function (event) {
   event.preventDefault();
   var $title = $entryForm.elements['picture-title'].value;
   var $url = $entryForm.elements['photo-url'].value;
   var $notes = $entryForm.elements.notes.value;
 
-  var entryForm = {
-    title: $title,
-    url: $url,
-    notes: $notes,
-    nextEntryId: data.nextEntryId,
-    entryId: entryId
-  };
-
-  entryId++;
-  data.entries.push(entryForm);
-  data.nextEntryId++;
-  $userPhoto.src = 'images/placeholder-image-square.jpg';
-  $entryForm.reset();
-
-  var prependEntry = renderEntries(entryForm);
-  $unorderedList.prepend(prependEntry);
-
+  if (data.editing !== null) {
+    for (let z = 0; z < data.entries.length; z++) {
+      if (data.entries[z].entryId === data.editing.entryId) {
+        var editedEntry = data.entries[z];
+        editedEntry.title = $title;
+        editedEntry.url = $url;
+        editedEntry.notes = $notes;
+        data.entries.splice(z, 1, editedEntry);
+      }
+    }
+  } else {
+    var entryForm = {
+      title: $title,
+      url: $url,
+      notes: $notes,
+      nextEntryId: data.nextEntryId,
+      entryId: data.nextEntryId - 1
+    };
+
+    data.entries.push(entryForm);
+    data.nextEntryId++;
+    $userPhoto.src = 'images/placeholder-image-square.jpg';
+    $entryForm.reset();
+
+    var prependEntry = renderEntries(entryForm);
+    $unorderedList.prepend(prependEntry);
+  }
+  data.editing = null;
   $dataViewEntries.className = 'view';
   $dataViewForm.className = 'hidden';
   data.view = 'entries';
-
+  location.reload();
```

Comment:
Using `location.reload()` is not the way to handle updating your entries list. Refreshing the page unexpectedly is bad practice and makes for a bad user experience. You need to use the new or updated entry to update the DOM directly.

Reply:
https://user-images.githubusercontent.com/88458577/132767921-4b864381-9a71-4401-a99b-c91061d81ce8.mp4